### DISCUSSION TITLE
Custom Controls in V6

### DISCUSSION THREAD

**Discussion by milonic** [fabricjs/fabric.js#9708] [Q&A]
Hi,

Does anybody have any info on how to create custom controls in V6? - similar to this: http://fabricjs.com/custom-control-render

**Comment by asturur** [comment]
Hey @milonic not much has changed.
You can subclass the control or assign different function to an existing one.
What does your control have to do?

**Reply by milonic** [reply]
Hi @asturur thanks for that I'll take a look at it and let you know if I need any help. I probably will.

Basically I just add to the existing controls - buttons for delete, change opacity and lock/unlock

**Comment by milonic** [comment]
Hi Guys,

Ok, I give in. I need help!

I've had a very good look at this, for the last few hours, and have figured createObjectDefaultFunction is a function containing the default controls that is referenced by a frozen Object called "Index" but cannot figure out which object I need to subclass.

So, let's say I have this:

```
deleteControl = new fabric.Control({
   x: 0.5,
   y: -0.5,
   offsetY: -16,
   offsetX: 16,
   cursorStyle: 'pointer',
   mouseUpHandler: deleteObject,
   render: renderIcon(deleteImg),
   cornerSize: 24
});
```

What do I do with it to get it to appear with the other controls?

In V5 you would just add it to the fabric.Object.prototype.controls object and it would just work but I can't see it being anywhere near the same in V6

Please help! :)

**Reply by ShaMan123** [reply]
myObjectInstance.controls.key = new MyControl()

**Reply by ShaMan123** [reply]
You should read the breaking changes thoroughly

**Reply by milonic** [reply]
Thanks @ShaMan123 - I have definitely been through the breaking changes but that's not something I've come accross thus far. It's kinda why I'd like to get onto helping with the docs, it's been quite a painful experience lol - good fun though.

**Reply by asturur** [reply]
The breaking changes are not even full described, i stopped annotating at some point, not sure if @ShaMan123 did more.

**Comment by Julian-B90** [comment]
All my Object Reac, Circle usw have a delete control but not my Drawing line with isDrawingMode=true. How i can add the Custom Control to freeDrawing?

**Reply by asturur** [reply]
https://fabricjs.github.io/docs/configuring-controls/

Here is a description on how you can work with custom controls per classes.
The drawing line while is being drawn has no controls and i m not sure they can be added
